contracts/test/integration/IntegrationTestVIP8.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {ICore} from "../../core/ICore.sol";
+import {IVolt} from "../../volt/Volt.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {vip8} from "./vip/vip8.sol";
+import {TimelockSimulation} from "./utils/TimelockSimulation.sol";
+import {PriceBoundPSM} from "../../peg/PriceBoundPSM.sol";
+import {IPCVGuardian} from "../../pcv/IPCVGuardian.sol";
+
+contract IntegrationTestVIP8 is TimelockSimulation, vip8 {
+    using SafeCast for *;
+    PriceBoundPSM private psm;
+
+    ICore private core = ICore(MainnetAddresses.CORE);
+    IERC20 dai = IERC20(MainnetAddresses.DAI);
+    IVolt volt = IVolt(MainnetAddresses.VOLT);
+
+    uint256 public constant mintAmount = 2_000_000e18;
+
+    function setUp() public {
+        psm = PriceBoundPSM(MainnetAddresses.VOLT_DAI_PSM);
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        /// mint VOLT to the user
+        volt.mint(address(psm), mintAmount);
+        volt.mint(address(this), mintAmount);
+        core.revokeMinter(MainnetAddresses.GOVERNOR);
+        vm.stopPrank();
+
+        vm.prank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(this), mintAmount);
+
+        mainnetSetup();
+        simulate(
+            getMainnetProposal(),
+            TimelockController(payable(MainnetAddresses.TIMELOCK_CONTROLLER)),
+            IPCVGuardian(MainnetAddresses.PCV_GUARDIAN),
+            MainnetAddresses.GOVERNOR,
+            MainnetAddresses.EOA_1,
+            vm,
+            false
+        );
+    }
+
+    function testRedeem(uint80 amountVoltIn) public {
```

Comment:
this will cap the input to 1.2m, we can change to uint96, which is a max of 79b to allow a larger search space